### ISSUE TITLE
System.Net.Security.Tests failed on armel

### ISSUE THREAD

**Issue by YongseopKim** [dotnet/corefx#21075]
System.Net.Security.Tests failed on armel

failed tcs are in tests/FunctionalTests/NegotiateStreamTestForUnix.cs
```
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_AuthToHttpTarget_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthDefaultCredentialsNoSeed_Failure [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_KerberosCreds_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthDefaultCredentials_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_EchoServer_NTLM_ClientWriteRead_Successive_Sync_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_EchoServer_ClientWriteRead_Successive_Sync_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthentication_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthInvalidPassword_Failure [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthInvalidUser_Failure [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "EST", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_EchoServer_NTLM_ClientWriteRead_Successive_Async_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_EchoServer_ClientWriteRead_Successive_Async_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_Fallback_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(credential: NetworkCredential { Domain = "TEST", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthWithoutRealm_Success [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "TEST", Password = "ntlm_password", SecurePaecureString { Length = 13 }, UserName = "tlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "TEST", Password = "tlm_password", SecurePassword = SecureString { Length = 12 }, UserName = "ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "EST", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "TEST\\ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user@EST" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_NtlmUser_InvalidCredentials_Fail(credential: NetworkCredential { Domain = "TEST.COREFX.NET", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(credential: NetworkCredential { Domain = "", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(credential: NetworkCredential { Domain = "", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "TEST\\ntlm_user" }) [FAIL]
System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(credential: NetworkCredential { Domain = "", Password = "ntlm_password", SecurePassword = SecureString { Length = 13 }, UserName = "ntlm_user@TEST" }) [FAIL]
```

There are two thrown exceptions:
```
   System.Net.Security.Tests.NegotiateStreamTest.NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure [FAIL]
      System.AggregateException : One or more errors occurred. (No such file or directory) (The following constructor parameters did not have matching fixture data: KDCSetup fixture)
      ---- System.ComponentModel.Win32Exception : No such file or directory
      ---- The following constructor parameters did not have matching fixture data: KDCSetup fixture
      Stack Trace:
         
         ----- Inner Stack Trace #1 (System.ComponentModel.Win32Exception) -----
            at System.Diagnostics.Process.ResolvePath(String filename)
            at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
            at System.Diagnostics.Process.Start()
            at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
         /home/dragon/tizen.release/corefx.release.2.0.0/src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTestForUnix.cs(111,0): at System.Net.Security.Tests.KDCSetup.RunSetupScript(String args)
         /home/dragon/tizen.release/corefx.release.2.0.0/src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTestForUnix.cs(42,0): at System.Net.Security.Tests.KDCSetup..ctor()
         ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----

      System.MissingMethodException : Constructor on type 'System.Net.Security.Tests.NegotiateStreamTest' not found.
      Stack Trace:
            at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)
            at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
            at System.Activator.CreateInstance(Type type, Object[] args)
            at ReflectionAbstractionExtensions.<>c__DisplayClass0_0.<CreateTestClass>b__0()
            at ReflectionAbstractionExtensions.CreateTestClass(ITest test, Type testClassType, Object[] constructorArguments, IMessageBus messageBus, ExecutionTimer timer, CancellationTokenSource cancellationTokenSource)
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

it's because of below code.
```cs
// in tests/FunctionalTests/NegotiateStreamTestForUnix.cs
namespace System.Net.Security.Tests
{
    public class KDCSetup : IDisposable
    {
        private static int RunSetupScript(string args = null)
        {
            ProcessStartInfo startInfo = new ProcessStartInfo();

            // since ProcessStartInfo does not support Verb, we use sudo as
            // the program to be run
            startInfo.FileName = SudoCommand;  // sudo
            startInfo.Arguments = string.Format("bash {0} {1}", ScriptName, args);  // bash setup-kdc.sh --password {guid} --yes

                                                        // sudo bash setup-kdc.sh --password {guid} --yes
            using (Process kdcSetup = Process.Start(startInfo))  // if sudo is not there, it throws System.ComponentModel.Win32Exception in ResolvePath() in Process.StartCore()
            {
                kdcSetup.WaitForExit();
                return kdcSetup.ExitCode;
            }
        }
```

Current rootfs of armel doesn't have `sudo`.

```
~/nfs/corefx/cross/rootfs (master ✘)✭ ᐅ ls
arm  armel  x86

~/nfs/corefx/cross/rootfs (master ✘)✭ ᐅ sudo find . -name "sudo"
./arm/var/lib/sudo
./arm/usr/share/doc/sudo
./arm/usr/share/lintian/overrides/sudo
./arm/usr/bin/sudo
./arm/usr/lib/sudo
./arm/etc/init.d/sudo
./arm/etc/pam.d/sudo
./x86/var/lib/sudo
./x86/usr/share/doc/sudo
./x86/usr/share/lintian/overrides/sudo
./x86/usr/bin/sudo
./x86/usr/lib/sudo
./x86/etc/init.d/sudo
./x86/etc/pam.d/sudo
```

This needs to care about a situation where there is no `sudo`.

**Comment by YongseopKim** [comment]
cc\ @hseok-oh

**Comment by davidsh** [comment]
Fixed by #21077.
